playground/backend/internal/db/datastore/datastore_db.go

```diff
@@ -429,39 +427,52 @@ func (d *Datastore) GetExampleGraph(ctx context.Context, id string) (string, err
 	return pcObj.Content, nil
 }
 
-//DeleteUnusedSnippets deletes all unused snippets
-func (d *Datastore) DeleteUnusedSnippets(ctx context.Context, dayDiff int32) error {
-	var hoursDiff = dayDiff * 24
-	boundaryDate := time.Now().Add(-time.Hour * time.Duration(hoursDiff))
-	snippetQuery := datastore.NewQuery(constants.SnippetKind).
-		Namespace(utils.GetNamespace(ctx)).
-		Filter("lVisited <= ", boundaryDate).
-		Filter("origin =", constants.UserSnippetOrigin).
+func (d *Datastore) deleteSnippets(ctx context.Context, snippetQuery *datastore.Query,
+	postQueryFilter func(*datastore.Key) bool) error {
```

Comment:
This filter is a little confusing since the things that are passing through the filter are not the things we're acting on (and deleting). Could you add an explanatory comment? And maybe rename to something like `skipDeletionFilter`? 

Alternately, this might just be clearer as a slice `skipKeys []*datastore.Key` instead of a function (that's probably an easier experience for the invoker as well).